lixPackageSets.git: 2.94.0-pre-20250704_362bfd827f52 -> 2.94.0-pre-20250711_ae00b1298353

alois31 opened the pull request:
<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [Nixpkgs 25.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/doc/release-notes/rl-2511.section.md) (or backporting [25.05](https://github.com/NixOS/nixpkgs/blob/master/doc/manual/release-notes/rl-2505.section.md) Nixpkgs Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
- [NixOS 25.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2511.section.md) (or backporting [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) NixOS Release notes)
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md), [pkgs/README.md](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md), [maintainers/README.md](https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md) and other contributing documentation in corresponding paths.

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

alois31 committed:
lixPackageSets.*.lix: omit unneeded Python packages when !doInstallCheck

alois31 committed:
lixPackageSets.git: 2.94.0-pre-20250704_362bfd827f52 -> 2.94.0-pre-20250807_8bbd5e1d0df9

lf- reviewed (commented):
hilarious bug. I've stamped the upstream fix so you could probably submit that and then upgrade past the bug to not have to deal with it here

alois31 commented:
Tests are still building, but I don't expect breakage since the source is the same as in the original and only the version number is different.

Aleksanaa commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha)

Command: `nixpkgs-review pr 424775`

Logs: https://github.com/Aleksanaa/nixpkgs-review-gha/actions/runs/16290703265


---
### `x86_64-linux` (sandbox = true)
<details>
  <summary>:x: 13 packages failed to build:</summary>
  <ul>
    <li>lixPackageSets.git.colmena</li>
    <li>lixPackageSets.git.lix</li>
    <li>lixPackageSets.git.lix.debug</li>
    <li>lixPackageSets.git.lix.dev</li>
    <li>lixPackageSets.git.lix.devdoc</li>
    <li>lixPackageSets.git.lix.doc</li>
    <li>lixPackageSets.git.lix.man</li>
    <li>lixPackageSets.git.nix-direnv</li>
    <li>lixPackageSets.git.nix-eval-jobs</li>
    <li>lixPackageSets.git.nix-fast-build</li>
    <li>lixPackageSets.git.nix-fast-build.dist</li>
    <li>lixPackageSets.git.nixpkgs-review</li>
    <li>lixPackageSets.git.nixpkgs-review.dist</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `x86_64-linux`</summary>
<details>
<summary>lixPackageSets.git.lix</summary>
<pre>[143/143] 🌖 lix:installcheck / functional2                                                          8/300s
[143/143] 🌗 lix:installcheck / functional2                                                          9/300s
[143/143] 🌘 lix:installcheck / functional2                                                         10/300s
[143/143] 🌑 lix:installcheck / functional2                                                         11/300s
[143/143] 🌒 lix:installcheck / functional2                                                         12/300s
[143/143] 🌓 lix:installcheck / functional2                                                         13/300s
[143/143] 🌔 lix:installcheck / functional2                                                         14/300s
[143/143] 🌕 lix:installcheck / functional2                                                         15/300s
[143/143] 🌖 lix:installcheck / functional2                                                         16/300s
[143/143] 🌗 lix:installcheck / functional2                                                         17/300s
[143/143] 🌘 lix:installcheck / functional2                                                         18/300s
[143/143] 🌑 lix:installcheck / functional2                                                         19/300s
[143/143] 🌒 lix:installcheck / functional2                                                         20/300s
143/143 lix:installcheck / functional2                                              OK              21.60s

Ok:                129 
Fail:              11  
Skipped:           3   

Full log written to /build/source/build/meson-logs/testlog.txt</pre>
</details>
</details>


---
### `aarch64-linux` (sandbox = true)
<details>
  <summary>:x: 13 packages failed to build:</summary>
  <ul>
    <li>lixPackageSets.git.colmena</li>
    <li>lixPackageSets.git.lix</li>
    <li>lixPackageSets.git.lix.debug</li>
    <li>lixPackageSets.git.lix.dev</li>
    <li>lixPackageSets.git.lix.devdoc</li>
    <li>lixPackageSets.git.lix.doc</li>
    <li>lixPackageSets.git.lix.man</li>
    <li>lixPackageSets.git.nix-direnv</li>
    <li>lixPackageSets.git.nix-eval-jobs</li>
    <li>lixPackageSets.git.nix-fast-build</li>
    <li>lixPackageSets.git.nix-fast-build.dist</li>
    <li>lixPackageSets.git.nixpkgs-review</li>
    <li>lixPackageSets.git.nixpkgs-review.dist</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `aarch64-linux`</summary>
<details>
<summary>lixPackageSets.git.lix</summary>
<pre>[142-143/143] 🌗 lix:installcheck / functional2                                                      5/300s
[142-143/143] 🌘 lix:installcheck / functional-binary-cache-build-remote                            16/300s
142/143 lix:installcheck / functional-binary-cache-build-remote                     OK              16.09s
[143/143] 🌑 lix:installcheck / functional2                                                          6/300s
[143/143] 🌒 lix:installcheck / functional2                                                          7/300s
[143/143] 🌓 lix:installcheck / functional2                                                          8/300s
[143/143] 🌔 lix:installcheck / functional2                                                          9/300s
[143/143] 🌕 lix:installcheck / functional2                                                         10/300s
[143/143] 🌖 lix:installcheck / functional2                                                         11/300s
[143/143] 🌗 lix:installcheck / functional2                                                         12/300s
[143/143] 🌘 lix:installcheck / functional2                                                         13/300s
[143/143] 🌑 lix:installcheck / functional2                                                         14/300s
[143/143] 🌒 lix:installcheck / functional2                                                         15/300s
143/143 lix:installcheck / functional2                                              OK              16.81s

Ok:                129 
Fail:              11  
Skipped:           3   

Full log written to /build/source/build/meson-logs/testlog.txt</pre>
</details>
</details>


---
### `x86_64-darwin` (sandbox = true)
<details>
  <summary>:white_check_mark: 12 packages built:</summary>
  <ul>
    <li>lixPackageSets.git.colmena</li>
    <li>lixPackageSets.git.lix</li>
    <li>lixPackageSets.git.lix.dev</li>
    <li>lixPackageSets.git.lix.devdoc</li>
    <li>lixPackageSets.git.lix.doc</li>
    <li>lixPackageSets.git.lix.man</li>
    <li>lixPackageSets.git.nix-direnv</li>
    <li>lixPackageSets.git.nix-eval-jobs</li>
    <li>lixPackageSets.git.nix-fast-build</li>
    <li>lixPackageSets.git.nix-fast-build.dist</li>
    <li>lixPackageSets.git.nixpkgs-review</li>
    <li>lixPackageSets.git.nixpkgs-review.dist</li>
  </ul>
</details>


---
### `aarch64-darwin` (sandbox = true)
<details>
  <summary>:white_check_mark: 12 packages built:</summary>
  <ul>
    <li>lixPackageSets.git.colmena</li>
    <li>lixPackageSets.git.lix</li>
    <li>lixPackageSets.git.lix.dev</li>
    <li>lixPackageSets.git.lix.devdoc</li>
    <li>lixPackageSets.git.lix.doc</li>
    <li>lixPackageSets.git.lix.man</li>
    <li>lixPackageSets.git.nix-direnv</li>
    <li>lixPackageSets.git.nix-eval-jobs</li>
    <li>lixPackageSets.git.nix-fast-build</li>
    <li>lixPackageSets.git.nix-fast-build.dist</li>
    <li>lixPackageSets.git.nixpkgs-review</li>
    <li>lixPackageSets.git.nixpkgs-review.dist</li>
  </ul>
</details>

alois31 committed:
capnproto: fix platform offset

Due to the wrong platform offset, buildPackages.capnproto would inappropriately
depend on empty-libgcc_eh when building with dynamic build platform and static
host platform. In turn the build would fail due to the archiver not being
found.

alois31 commented:
Static build still doesn't work due to cmake failure when trying to find toml11.

alois31 commented:
Marking as ready for review again since the failure already happens on the base so this update doesn't make anything worse. Of course it would still be good to fix the issue. Relevant excerpt from the CMake logs embedded in the Meson logs:

```
CMake binary for host machine is cached.
Extracting basic cmake information
CMake Toolchain: Calling CMake once to generate the compiler state
Calling CMake (['/nix/store/hvn9cprr3p61gjlahk9pq4bmp3zaxj4a-cmake-3.31.7/bin/cmake']) in /build/source/build/meson-private/__CMake_compiler_info__ with:
  - "--trace-expand"
  - "--trace-format=json-v1"
  - "--no-warn-unused-cli"
  - "--trace-redirect=cmake_trace.txt"
  - "-G"
  - "Ninja"
  - "-DCMAKE_TOOLCHAIN_FILE=/build/source/build/meson-private/__CMake_compiler_info__/CMakeMesonTempToolchainFile.cmake"
  - "."
WARNING: CMake Toolchain: Failed to determine CMake compilers state
 -- return code: 1
 -- stdout: Put cmake in trace mode, but with variables expanded.
 -- stdout: Put cmake in trace mode and sets the trace output format.
 -- stdout: Not searching for unused variables given on the command line.
 -- stdout: Put cmake in trace mode and redirect trace output to a file instead of stderr.
 -- stdout: Trace will be written to cmake_trace.txt
 -- stdout: -- The CXX compiler identification is Clang 19.1.7
 -- stdout: -- Detecting CXX compiler ABI info
 -- stdout: -- Detecting CXX compiler ABI info - failed
 -- stdout: -- Check for working CXX compiler: /nix/store/51gyhiwsp1lxkzfa5649b9piizq8fsgh-x86_64-unknown-linux-musl-clang-wrapper-19.1.7/bin/x86_64-unknown-linux-musl-clang++
 -- stdout: -- Check for working CXX compiler: /nix/store/51gyhiwsp1lxkzfa5649b9piizq8fsgh-x86_64-unknown-linux-musl-clang-wrapper-19.1.7/bin/x86_64-unknown-linux-musl-clang++ - broken
 -- stdout: -- Configuring incomplete, errors occurred!
 -- stdout: 
 -- stderr: CMake Warning (dev) at /nix/store/hvn9cprr3p61gjlahk9pq4bmp3zaxj4a-cmake-3.31.7/share/cmake-3.31/Modules/CMakeDetermineCXXCompiler.cmake:74 (mark_as_advanced):
 -- stderr:   Policy CMP0102 is not set: The variable named "CMAKE_CXX_COMPILER" is not
 -- stderr:   in the cache.  This results in an empty cache entry which is no longer
 -- stderr:   created when policy CMP0102 is set to NEW.  Run "cmake --help-policy
 -- stderr:   CMP0102" for policy details.  Use the cmake_policy command to set the
 -- stderr:   policy and suppress this warning.
 -- stderr: Call Stack (most recent call first):
 -- stderr:   CMakeLists.txt:3 (project)
 -- stderr: This warning is for project developers.  Use -Wno-dev to suppress it.
 -- stderr: 
 -- stderr: CMake Error at /nix/store/hvn9cprr3p61gjlahk9pq4bmp3zaxj4a-cmake-3.31.7/share/cmake-3.31/Modules/CMakeTestCXXCompiler.cmake:73 (message):
 -- stderr:   The C++ compiler
 -- stderr: 
 -- stderr:     "/nix/store/51gyhiwsp1lxkzfa5649b9piizq8fsgh-x86_64-unknown-linux-musl-clang-wrapper-19.1.7/bin/x86_64-unknown-linux-musl-clang++"
 -- stderr: 
 -- stderr:   is not able to compile a simple test program.
 -- stderr: 
 -- stderr:   It fails with the following output:
 -- stderr: 
 -- stderr:     Change Dir: '/build/source/build/meson-private/__CMake_compiler_info__/CMakeFiles/CMakeScratch/TryCompile-KP6xlj'
 -- stderr: 
 -- stderr:     Run Build Command(s): /nix/store/ac8gl7n529sp9684hwxpsbmjbgpjkvf4-ninja-1.12.1/bin/ninja -v cmTC_bd1e3
 -- stderr:     [1/2] /nix/store/51gyhiwsp1lxkzfa5649b9piizq8fsgh-x86_64-unknown-linux-musl-clang-wrapper-19.1.7/bin/x86_64-unknown-linux-musl-clang++    -MD -MT CMakeFiles/cmTC_bd1e3.dir/testCXXCompiler.cxx.o -MF CMakeFiles/cmTC_bd1e3.dir/testCXXCompiler.cxx.o.d -o CMakeFiles/cmTC_bd1e3.dir/testCXXCompiler.cxx.o -c /build/source/build/meson-private/__CMake_compiler_info__/CMakeFiles/CMakeScratch/TryCompile-KP6xlj/testCXXCompiler.cxx
 -- stderr:     clang++: warning: argument unused during compilation: '-pie' [-Wunused-command-line-argument]
 -- stderr:     [2/2] : && /nix/store/51gyhiwsp1lxkzfa5649b9piizq8fsgh-x86_64-unknown-linux-musl-clang-wrapper-19.1.7/bin/x86_64-unknown-linux-musl-clang++   CMakeFiles/cmTC_bd1e3.dir/testCXXCompiler.cxx.o -o cmTC_bd1e3   && :
 -- stderr:     FAILED: cmTC_bd1e3
 -- stderr:     : && /nix/store/51gyhiwsp1lxkzfa5649b9piizq8fsgh-x86_64-unknown-linux-musl-clang-wrapper-19.1.7/bin/x86_64-unknown-linux-musl-clang++   CMakeFiles/cmTC_bd1e3.dir/testCXXCompiler.cxx.o -o cmTC_bd1e3   && :
 -- stderr:     /nix/store/jncwlw2mqkap6baffah6d7ihfqrrrnb8-x86_64-unknown-linux-musl-binutils-2.44/bin/x86_64-unknown-linux-musl-ld: warning: libc.so.6, needed by /nix/store/fxwmabprl1nyxskbrygqlaps5kwg3fjv-gcc-14.3.0-libgcc/lib/libgcc_s.so.1, not found (try using -rpath or -rpath-link)
 -- stderr:     /nix/store/jncwlw2mqkap6baffah6d7ihfqrrrnb8-x86_64-unknown-linux-musl-binutils-2.44/bin/x86_64-unknown-linux-musl-ld: warning: ld-linux-x86-64.so.2, needed by /nix/store/fxwmabprl1nyxskbrygqlaps5kwg3fjv-gcc-14.3.0-libgcc/lib/libgcc_s.so.1, not found (try using -rpath or -rpath-link)
 -- stderr:     /nix/store/jncwlw2mqkap6baffah6d7ihfqrrrnb8-x86_64-unknown-linux-musl-binutils-2.44/bin/x86_64-unknown-linux-musl-ld: /nix/store/fxwmabprl1nyxskbrygqlaps5kwg3fjv-gcc-14.3.0-libgcc/lib/libgcc_s.so.1: undefined reference to `memset@GLIBC_2.2.5'
 -- stderr:     /nix/store/jncwlw2mqkap6baffah6d7ihfqrrrnb8-x86_64-unknown-linux-musl-binutils-2.44/bin/x86_64-unknown-linux-musl-ld: /nix/store/fxwmabprl1nyxskbrygqlaps5kwg3fjv-gcc-14.3.0-libgcc/lib/libgcc_s.so.1: undefined reference to `strlen@GLIBC_2.2.5'
 -- stderr:     /nix/store/jncwlw2mqkap6baffah6d7ihfqrrrnb8-x86_64-unknown-linux-musl-binutils-2.44/bin/x86_64-unknown-linux-musl-ld: /nix/store/fxwmabprl1nyxskbrygqlaps5kwg3fjv-gcc-14.3.0-libgcc/lib/libgcc_s.so.1: undefined reference to `pthread_setspecific@GLIBC_2.34'
 -- stderr:     /nix/store/jncwlw2mqkap6baffah6d7ihfqrrrnb8-x86_64-unknown-linux-musl-binutils-2.44/bin/x86_64-unknown-linux-musl-ld: /nix/store/fxwmabprl1nyxskbrygqlaps5kwg3fjv-gcc-14.3.0-libgcc/lib/libgcc_s.so.1: undefined reference to `mmap@GLIBC_2.2.5'
 -- stderr:     /nix/store/jncwlw2mqkap6baffah6d7ihfqrrrnb8-x86_64-unknown-linux-musl-binutils-2.44/bin/x86_64-unknown-linux-musl-ld: /nix/store/fxwmabprl1nyxskbrygqlaps5kwg3fjv-gcc-14.3.0-libgcc/lib/libgcc_s.so.1: undefined reference to `pthread_cond_broadcast@GLIBC_2.3.2'
 -- stderr:     /nix/store/jncwlw2mqkap6baffah6d7ihfqrrrnb8-x86_64-unknown-linux-musl-binutils-2.44/bin/x86_64-unknown-linux-musl-ld: /nix/store/fxwmabprl1nyxskbrygqlaps5kwg3fjv-gcc-14.3.0-libgcc/lib/libgcc_s.so.1: undefined reference to `pthread_getspecific@GLIBC_2.34'
 -- stderr:     /nix/store/jncwlw2mqkap6baffah6d7ihfqrrrnb8-x86_64-unknown-linux-musl-binutils-2.44/bin/x86_64-unknown-linux-musl-ld: /nix/store/fxwmabprl1nyxskbrygqlaps5kwg3fjv-gcc-14.3.0-libgcc/lib/libgcc_s.so.1: undefined reference to `pthread_once@GLIBC_2.34'
 -- stderr:     /nix/store/jncwlw2mqkap6baffah6d7ihfqrrrnb8-x86_64-unknown-linux-musl-binutils-2.44/bin/x86_64-unknown-linux-musl-ld: /nix/store/fxwmabprl1nyxskbrygqlaps5kwg3fjv-gcc-14.3.0-libgcc/lib/libgcc_s.so.1: undefined reference to `getpagesize@GLIBC_2.2.5'
 -- stderr:     /nix/store/jncwlw2mqkap6baffah6d7ihfqrrrnb8-x86_64-unknown-linux-musl-binutils-2.44/bin/x86_64-unknown-linux-musl-ld: /nix/store/fxwmabprl1nyxskbrygqlaps5kwg3fjv-gcc-14.3.0-libgcc/lib/libgcc_s.so.1: undefined reference to `pthread_mutex_lock@GLIBC_2.2.5'
 -- stderr:     /nix/store/jncwlw2mqkap6baffah6d7ihfqrrrnb8-x86_64-unknown-linux-musl-binutils-2.44/bin/x86_64-unknown-linux-musl-ld: /nix/store/fxwmabprl1nyxskbrygqlaps5kwg3fjv-gcc-14.3.0-libgcc/lib/libgcc_s.so.1: undefined reference to `realloc@GLIBC_2.2.5'
 -- stderr:     /nix/store/jncwlw2mqkap6baffah6d7ihfqrrrnb8-x86_64-unknown-linux-musl-binutils-2.44/bin/x86_64-unknown-linux-musl-ld: /nix/store/fxwmabprl1nyxskbrygqlaps5kwg3fjv-gcc-14.3.0-libgcc/lib/libgcc_s.so.1: undefined reference to `pthread_cond_wait@GLIBC_2.3.2'
 -- stderr:     /nix/store/jncwlw2mqkap6baffah6d7ihfqrrrnb8-x86_64-unknown-linux-musl-binutils-2.44/bin/x86_64-unknown-linux-musl-ld: /nix/store/fxwmabprl1nyxskbrygqlaps5kwg3fjv-gcc-14.3.0-libgcc/lib/libgcc_s.so.1: undefined reference to `calloc@GLIBC_2.2.5'
 -- stderr:     /nix/store/jncwlw2mqkap6baffah6d7ihfqrrrnb8-x86_64-unknown-linux-musl-binutils-2.44/bin/x86_64-unknown-linux-musl-ld: /nix/store/fxwmabprl1nyxskbrygqlaps5kwg3fjv-gcc-14.3.0-libgcc/lib/libgcc_s.so.1: undefined reference to `pthread_mutex_unlock@GLIBC_2.2.5'
 -- stderr:     /nix/store/jncwlw2mqkap6baffah6d7ihfqrrrnb8-x86_64-unknown-linux-musl-binutils-2.44/bin/x86_64-unknown-linux-musl-ld: /nix/store/fxwmabprl1nyxskbrygqlaps5kwg3fjv-gcc-14.3.0-libgcc/lib/libgcc_s.so.1: undefined reference to `malloc@GLIBC_2.2.5'
 -- stderr:     /nix/store/jncwlw2mqkap6baffah6d7ihfqrrrnb8-x86_64-unknown-linux-musl-binutils-2.44/bin/x86_64-unknown-linux-musl-ld: /nix/store/fxwmabprl1nyxskbrygqlaps5kwg3fjv-gcc-14.3.0-libgcc/lib/libgcc_s.so.1: undefined reference to `pthread_key_create@GLIBC_2.34'
 -- stderr:     /nix/store/jncwlw2mqkap6baffah6d7ihfqrrrnb8-x86_64-unknown-linux-musl-binutils-2.44/bin/x86_64-unknown-linux-musl-ld: /nix/store/fxwmabprl1nyxskbrygqlaps5kwg3fjv-gcc-14.3.0-libgcc/lib/libgcc_s.so.1: undefined reference to `__tls_get_addr@GLIBC_2.3'
 -- stderr:     /nix/store/jncwlw2mqkap6baffah6d7ihfqrrrnb8-x86_64-unknown-linux-musl-binutils-2.44/bin/x86_64-unknown-linux-musl-ld: /nix/store/fxwmabprl1nyxskbrygqlaps5kwg3fjv-gcc-14.3.0-libgcc/lib/libgcc_s.so.1: undefined reference to `memmove@GLIBC_2.2.5'
 -- stderr:     /nix/store/jncwlw2mqkap6baffah6d7ihfqrrrnb8-x86_64-unknown-linux-musl-binutils-2.44/bin/x86_64-unknown-linux-musl-ld: /nix/store/fxwmabprl1nyxskbrygqlaps5kwg3fjv-gcc-14.3.0-libgcc/lib/libgcc_s.so.1: undefined reference to `abort@GLIBC_2.2.5'
 -- stderr:     /nix/store/jncwlw2mqkap6baffah6d7ihfqrrrnb8-x86_64-unknown-linux-musl-binutils-2.44/bin/x86_64-unknown-linux-musl-ld: /nix/store/fxwmabprl1nyxskbrygqlaps5kwg3fjv-gcc-14.3.0-libgcc/lib/libgcc_s.so.1: undefined reference to `memcpy@GLIBC_2.14'
 -- stderr:     /nix/store/jncwlw2mqkap6baffah6d7ihfqrrrnb8-x86_64-unknown-linux-musl-binutils-2.44/bin/x86_64-unknown-linux-musl-ld: /nix/store/fxwmabprl1nyxskbrygqlaps5kwg3fjv-gcc-14.3.0-libgcc/lib/libgcc_s.so.1: undefined reference to `munmap@GLIBC_2.2.5'
 -- stderr:     /nix/store/jncwlw2mqkap6baffah6d7ihfqrrrnb8-x86_64-unknown-linux-musl-binutils-2.44/bin/x86_64-unknown-linux-musl-ld: /nix/store/fxwmabprl1nyxskbrygqlaps5kwg3fjv-gcc-14.3.0-libgcc/lib/libgcc_s.so.1: undefined reference to `free@GLIBC_2.2.5'
 -- stderr:     /nix/store/jncwlw2mqkap6baffah6d7ihfqrrrnb8-x86_64-unknown-linux-musl-binutils-2.44/bin/x86_64-unknown-linux-musl-ld: /nix/store/fxwmabprl1nyxskbrygqlaps5kwg3fjv-gcc-14.3.0-libgcc/lib/libgcc_s.so.1: undefined reference to `_dl_find_object@GLIBC_2.35'
 -- stderr:     clang++: error: linker command failed with exit code 1 (use -v to see invocation)
 -- stderr:     ninja: build stopped: subcommand failed.
 -- stderr: 
 -- stderr: 
 -- stderr: 
 -- stderr: 
 -- stderr: 
 -- stderr:   CMake will not be able to correctly generate this project.
 -- stderr: Call Stack (most recent call first):
 -- stderr:   CMakeLists.txt:3 (project)
 -- stderr: 
 -- stderr: 
 -- stderr: 
```

It seems like somehow it tries to link against the wrong libgcc.

alois31 commented:
Bisected with `--first-parent` to the last staging-next:
```
b47d4f01d4213715a1f09b999bab96bb6a5b675e is the first bad commit
commit b47d4f01d4213715a1f09b999bab96bb6a5b675e
Merge: b2e5044b3e79 a82a7e5045b5
Author: Vladimír Čunát <v@cunat.cz>
Date:   Sun Jul 13 19:33:14 2025 +0200

    staging-next 2025-07-04 (#422427)
```

lf- reviewed (approved):
overall lgtm, confused at why we want to skip structured attrs

RaitoBezarius commented:
@symphorien could we have your help for the static builds as you landed the structured attrs change? You may have a hint to what's going on.

alois31 commented on pkgs/tools/package-management/lix/common-lix.nix:
```diff
@@ -131,20 +132,26 @@ stdenv.mkDerivation (finalAttrs: {
     stdenv.cc.cc.stdenv.cc.cc
   ];
-  __structuredAttrs = true;
+  __structuredAttrs = finalAttrs.separateDebugInfo;
```
Because it is the thing that broke static builds. I will continue debugging this.

RaitoBezarius replied:
static builds

lf- reviewed (changes requested):
needs comment on structured attrs

alois31 commented:
Some progress on the `__structuredAttrs` issue: apparently the difference that matters is that `NIX_CFLAGS_COMPILE=" -static"` no longer leaks. Still figuring out the best way to fix this.

alois31 committed:
lixPackageSets.*.lix: stop copying unnecessary boost libraries

This is basically the same as https://gerrit.lix.systems/c/lix/+/3767 upstream.

alois31 committed:
lixPackageSets.*.lix: use buildPackages.python3 instead of python3.pythonOnBuildForHost

The former seems to exhibit fewer subtle issues breaking cross (in particular
static) builds, for reasons I have not yet understood.

alois31 commented:
`buildPackages.python3` seems to work better than `python3.pythonOnBuildForHost`. Why? I have no idea. It's also not perfect because it's not affected by the issue leading to https://github.com/NixOS/nixpkgs/pull/430368 even though it should, but that does not seem to actually be relevant in practice (and is a fortunate accident here).

philiptaron commented:
Is this ready to merge?

alois31 commented:
Static build doesn't work again due to kj fibers, but again it was broken before.
